torchmetrics/text/wip.py

```diff
@@ -0,0 +1,105 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, List, Optional, Union
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.text.wip import _wip_compute, _wip_update
+from torchmetrics.metric import Metric
+
+
+class WordInfoPreserved(Metric):
+    r"""
+    word Information Preserved (WordInfoPreserved_) is a metric of the performance of an automatic speech
+    recognition system. This value indicates the percentage of words that were correctly predicted between
+    a set of ground-truth sentences and a set of hypothesis sentences.
+    The higher the value, the better the performance of the ASR system with a WordInfoPreserved of 0
+    being a perfect score.
+    word Information Preserved rate can then be computed as:
+
+    .. math::
+        wip = \frac{C}{N} + \frac{C}{P}
+
+    where:
+
+        - C is the number of correct words,
+        - N is the number of words in the reference
+        - P is the number of words in the prediction
+
+
+    Args:
+        compute_on_step:
+            Forward only calls ``update()`` and return None if this is set to False.
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called. default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state. When ``None``, DDP
+            will be used to perform the allgather
+
+    Returns:
+        word Information Preserved score
```

Comment:
this docstring belongs to the init method which does not return anything. This should just be in the compute method (which it already is) :]
```suggestion
```